src/server/services/graph/createConversionArrays.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Unit = require('../../models/Unit');
+const { getPath } = require('./createConversionGraph');
+const { pathConversion } = require('./pathConversion');
+
+/**
+ * For each unit in the list, assigns its index to the conversion table.
+ * For source units, the index is the row id.
+ * For destination units, the index is the column id.
+ * @param {*} units The list of units.
+ * @param {*} conn The connection to use.
+ */
+async function assignIndex(units, conn) {
+	let id = 0;
+	// The old indices may cause conflicts while we are assigning new ones.
+	// Therefore, we need to reset them to null first.
+	for (let unit of units) {
```

Comment:
The design always had an issue that while Cik was being recreated any requests might return incorrect values. This was true until the DB settled and the new Cik was transferred to a client. Since this was rare we let it go. The concern I have here is that setting the index to null will cause an error if a client happens to use this value. I'm not sure that is worse than a wrong value but it might cause problems in the code. We need to think about what to do in that code. Maybe catch the exception, wait a second, and reload Cik. I don't think this is an issue to fix in this code but noting for future code to be added. Maybe a TODO comment here is desirable so we remember this.

Reply:
During the process of assigning new values, the old ids and new ids may be similar. This will cause an error since we have a constraint of unique unit index in the DB. So I think it's best to set everything to null. I agree with you that we should be careful during the recreating Cik array process. I've added a TODO comment about this.